src/sql/src/plan/statement/ddl.rs

```diff
@@ -1348,6 +1365,23 @@ pub fn plan_create_sink(
     })
 }
 
+/// Returns only those `CatalogItem`s that don't have any other user
+/// dependencies. Those are the root dependencies.
```

Comment:
so what happens if we make a view built on top of system tables / system logs?

Reply:
ah i think i see now -- we bail iff theres a dep thats not a source, or is a source but doesnt stably return messages